features/kafka/producer/src/main/proto/opennms-kafka-producer.proto

```diff
@@ -147,3 +147,68 @@ message Node {
   repeated SnmpInterface snmp_interface = 12;
   HwEntity hw_inventory = 13;
 }
+
+enum OnmsTopologyMessageStatus {
+  NEW = 0;
+  UPDATE=1;
+  DELETE=2;
+}
+
+message OnmsTopologyRef {
+  string id = 1;
+}
+
+message OnmsTopologyMessage {
```

Comment:
Can we remove the "Onms" prefix and the "Message" suffix from these types? All of the objects are implicitly Onms objects.

Reply:
Sure!


> I think we need to rework the way the objects are persisted in Kafka here to make it easier to consume.
> 
> Instead of requiring the consumer to keep track of the adds/updates/deletes, we can split the topology into two topics - one for vertices and one for edges. Vertices and edges have unique ids, so they should be keyed by these ids in the topic. When pushing updates, we simply push a new record over the same key. When pushing a delete, we push a null record over the key.
> 
> In this fashion, we can model the topology as a ktable similar to what we do with alarms.

I see but I have not a clear understanding of what you are just talking about.
Every topology Object has an unique id so no problem for implementng this logic

the code now creates a topic for each protocol CDP, LLDP, OSPF, BRIDGE and ISIS....then I send the refid....and then I serialize the object.

if the topology object is deleted I can send null.
But consider that a topology is a complex object...and edge (better shared) is a relation among vertices....in any case you have to rebuild topology....

Reply:
..we need a Topic for ev ery protocol vertex,edge and segment

a vertex is either a cdp and ospf vertex....then for some reason...ospf is not more supported..then you got a delete for the vertex in cdp....

if you have only one matrix with vertex ref....you will delete the vertex....so you need to distinguish among protocols.